### DISCUSSION TITLE
Existing installation with Blade - Can't display page content

### DISCUSSION THREAD

**Discussion by flymke** [statamic/cms#4386] [Q&A] [answered]
Hi there, I am new to statamic and found it to be a wonderful project.
I am currently playing around with an existing laravel app based on standard laravel routing (web.php, default controllers and blade views).

From what I read everywhere it should be easy to display the statamic content inside my blade views as well, e.g. {{ $title }}.
However when I try, I get always get **Undefined variable: title**

Only if I do this:
```
@php
$entry = \Statamic\Facades\Entry::findByUri('/home');  
@endphp
{{ $entry->get('title') }}
```

it works.

I already created a /home page in statamic. So my thought was that Statamic automagically wires the URLs to the correct page and injects the data into the view. Maybe I am wrong? Or there is something else missing yet?

I am thankful for any hints!

PS: I guess my question is: If I use standard laravel routes and controllers. Do I have to inject the data myself (within the controller or view) or is there maybe a better way for it?

**Comment by jasonvarga** [comment, marked as answer]
It's only automatic when you use Statamic's routing. If you're rolling your own controllers and views, you have to provide the variables.

**Reply by flymke** [reply]
Thanks for this answer!

**Reply by flymke** [reply]
Just to be sure: 
So, if I use Statamic routing, can I use my default laravel controllers then?
Or alternatively, how would you normally add controller logic to sites then?
Maybe I am thinking to much in my vanilla laravel box. :-)

**Reply by duncanmcclean** [reply]
What sort of extra controller logic are you thinking about? (there's most likely a way you can do it without making your own controllers to serve Statamic content)

**Reply by flymke** [reply]
Hm, I am thinking about custom functions that happen on a page, e.g. pull data from an external database or calculate things before sending them to the view. Things I normally would do inside my controller?

**Reply by duncanmcclean** [reply]
You could maybe use a [view model](https://statamic.dev/view-models#content) or some [custom tags](https://statamic.dev/extending/tags)?

**Reply by flymke** [reply]
Fantastic. Thank you. This is what I had in mind.
